### ISSUE TITLE
Visual Grounding, Whole sentence visualization in Fig.4

### ISSUE THREAD

**Issue by zzzzzigzag** [salesforce/ALBEF#51]
Hi, I tried to visualize the whole sentence attention map by [CLS], but the result is inaccurate.
For example: text tokens are [[CLS], "grey", "sweater"], the "grey" "sweater" tokens fits well in the image, but the [CLS] token heat map is incorrect. How did you get the result of Figure 4? Thank you!

I've removed text attention mask:
```python
    if show_CLS:
        cams = cams[:, :, :, 1:].reshape(image.size(0), 12, -1, 24, 24)
        grads = grads[:, :, :, 1:].clamp(0).reshape(image.size(0), 12, -1, 24, 24)
    else:
        cams = cams[:, :, :, 1:].reshape(image.size(0), 12, -1, 24, 24) * mask
        grads = grads[:, :, :, 1:].clamp(0).reshape(image.size(0), 12, -1, 24, 24) * mask
```

**Comment by LiJunnan1992** [comment]
Hi, Figure 4 is computed by averaging all the token's heatmaps, as shown in Grounding.py:

```
  with torch.no_grad():                                 
      mask = text_input.attention_mask.view(text_input.attention_mask.size(0),1,-1,1,1)
      
      grads = model.text_encoder.base_model.base_model.encoder.layer[block_num].crossattention.self.get_attn_gradients().detach()
      cams = model.text_encoder.base_model.base_model.encoder.layer[block_num].crossattention.self.get_attention_map().detach()

      cams = cams[:, :, :, 1:].reshape(image.size(0), 12, -1, 24, 24) * mask              
      grads = grads[:, :, :, 1:].clamp(min=0).reshape(image.size(0), 12, -1, 24, 24) * mask
      
      gradcam = cams * grads
      gradcam = gradcam.mean(1).mean(1)
```

**Comment by zzzzzigzag** [comment]
Thanks for your reply!
